vpc/main.tf

```diff
@@ -35,23 +35,23 @@ resource "aws_vpc" "main" {
 }
 
 resource "aws_eip" "nat" {
-  count = var.enable_eip ? local.max_subnet_length : 0
+  count = var.enable_eip ? local.nat_gateway_count : 0
   # checkov:skip=CKV2_AWS_19:EIP is used by NAT Gateway
   vpc = true
   tags = merge(local.common_tags, {
-    Name = "${var.name}-eip${count.index}"
+    Name = var.single_nat_gateway ? "${var.name}-eip" : "${var.name}-eip${count.index}"
   })
 }
 
 resource "aws_nat_gateway" "nat_gw" {
-  count = local.max_subnet_length
+  count = local.max_subnet_length > 0 ? local.nat_gateway_count : 0
```

Comment:
Could this just be the `local.nat_gateway_count`?

Reply:
i like it